hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -682,6 +736,25 @@ protected InodeTree(final Configuration config, final String viewName,
     }
   }
 
+  private Collection<LinkEntry> getLinkEntries(List<LinkEntry> linkEntries) {
+    if (isNestedMountPointSupported) {
+      Set<LinkEntry> sortedLinkEntries = new TreeSet<>((o1, o2) -> {
+        if (o1 == null) {
+          return -1;
+        }
+        if (o2 == null) {
+          return 1;
+        }
+        String src1 = o1.getSrc().toLowerCase();
```

Comment:
Also, is lower casing the right way to do this? shouldn't you just compare the input Strings as /Foo/Bar and /foo/bar are different paths?

Reply:
Good catch, changed.